### PR TITLE
Changes to AI profile to accommodate expressing energyConsumption

### PR DESCRIPTION
**Pull request by rgopikrishnan91**
@kestewart @goneall This is part 4 of the PR. Please review it. With this the changes we discussed for Energy consumption should be done.

### CODE REVIEW

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+List different acceptable units of energy consumption. If the unit in which the energy consumption is recorded is not available here, please select other
+
+## Metadata
+
+- name: SafetyRiskAssessmentType
+
+## Entries
+
+- kW-H: Kilo Watt/ hour.
+- MJ: Mega Joule.
```
Comment (bact) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:20]: "MJ: Mega Joule." -> "MJ: Megajoule."

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+List different acceptable units of energy consumption. If the unit in which the energy consumption is recorded is not available here, please select other
+
+## Metadata
+
+- name: SafetyRiskAssessmentType
+
+## Entries
+
+- kW-H: Kilo Watt/ hour.
```
Comment (bact) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:19]: "kW-H: Kilo Watt/ hour."
->
"kWh: Kilowatt-hour." (lowercase "h")

Note that NIST suggested "kW · h" or "kW h" but that's probably too difficult to deal with as an entry name. See https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication811e2008.pdf (Notes 2 of the section 6.1.5 at page 12).

We can just use a more common "kWh" (NIST also recognized this practice, but not recommended).

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 1 above)
Comment (goneall) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:20]:
```suggestion
- mJ: Mega Joule.
```

To be consistent with naming conventions

**Review by goneall — CHANGES_REQUESTED** [review]
Couple changes to be consistent with the naming conventions

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 1 above)
Comment (bact) [reply model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:20]: Understand the consistency requirement, but that will made it read as "millijoule".

Comment (bact) [reply model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:20]: Can we avoid the confusing by using the full name `megajoule` as an entry name ?

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+# EnergyConsumptionDescriptionUnitType
+
+## Summary
+
+Specifies the unit of energy consumption
+
+## Description
+
+List different acceptable units of energy consumption. If the unit in which the energy consumption is recorded is not available here, please select other
```
Comment (kestewart) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:11]:
```suggestion
List different acceptable units of energy consumption. If the unit in which the energy consumption has been recorded in, is not available here, please select other.
```

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Metadata
+
+- name: SafetyRiskAssessmentType
```
Comment (kestewart) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:15]: uh?   I think this a copy paste error.
```suggestion
- name: EnergyConsumptionDescriptionUnitType
```

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 2 above)
Comment (kestewart) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:19]:
```suggestion
- kWh: Kilo Watt/ hour.
```

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 1 above)
Comment (kestewart) [reply model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:20]: good point.   +1 on just using full name to avoid ambiguity.

**Review by kestewart — CHANGES_REQUESTED** [review]

**Comment by rgopikrishnan91** [comment]
Signed-off-by: Gopi Krishnan Rajbahadur <gopikrishnanrajbahadur@gmail.com>

**Comment by rgopikrishnan91** [comment]
@bact @goneall @kestewart Please take a look at this PR. I have made the requested changes and combined all the changes here. Please let me know your thoughts.

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+List different acceptable units of energy consumption. If the unit in which the energy consumption has been recorded is not available here, please select other.
+
+## Metadata
+
+- name: EnergyConsumptionDescriptionUnitType
+
+## Entries
+
+- kilowatt-hour: KiloWatt per hour
```
Comment (bact) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:19]: Thank you @rgopikrishnan91 . Two things here:

1) The entry name `kilowatt-hour` may not follow the naming convention (as it contains `-` dash). We may need to change it to either `kWh` or `kilowatthour` (the latter is following the spelling in the glossary of U.S. Energy Information Administration). Also when watt is written in full, w is lowercase (except when it starts the sentence), but its symbol is an uppercase W.

2) The entry definition may need to be something along the line of "Kilowatt for one hour" and not "Kilowatt per hour". See my comment at [#discussion_r1552407399](https://github.com/spdx/spdx-3-model/pull/697#discussion_r1552407399). 

If we prefer conciseness, we can have something like `kWh: Kilowatt-hour`.

Refs:
- [Glossary - U.S. Energy Information Administration](https://www.eia.gov/tools/glossary/index.php?id=K)
- [kilowatt-hour - Wikipedia](https://en.wikipedia.org/wiki/Kilowatt-hour)

File: model/AI/Classes/EnergyConsumption
Comment (goneall) [inline model/AI/Classes/EnergyConsumption:1]: Missing the .md file extension on the file name

File: model/AI/Classes/EnergyConsumptionDescription
Comment (goneall) [inline model/AI/Classes/EnergyConsumptionDescription:1]: Same comment - add `.md`

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType
Comment (goneall) [inline model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:1]: Same comment - add `.md`

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 5 above)
Comment (goneall) [reply model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:19]: would it be `kilowattHour`?  uppercase H - otherwise agree with the suggestion

**Review by goneall — CHANGES_REQUESTED** [review]
A few file renamings requested.  Note - the renamed files will trigger a more robust CI checking.

File: model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType (hunk 5 above)
Comment (bact) [reply model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType:19]: We can go with `kilowattHour: Kilowatt-hour.` as well.

File: model/AI/Classes/EnergyConsumption [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# EnergyConsumption
+
+## Summary
+
+Class that contains properties to describe energy consumption incurred by a AI model in different stages of its lifecycle.
+
+## Description
+
+The class used for denoting the training energy consumption, inference energy consumption and fine tuning energy consumption of the AI model(s) used in a AI system.
```
Comment (bact) [inline model/AI/Classes/EnergyConsumption:11]: "a AI system" -> "an AI system"

File: model/AI/Classes/EnergyConsumption [hunk 7]
```diff
@@ -0,0 +1,25 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# EnergyConsumption
+
+## Summary
+
+Class that contains properties to describe energy consumption incurred by a AI model in different stages of its lifecycle.
```
Comment (bact) [inline model/AI/Classes/EnergyConsumption:7]: "a AI model" -> "an AI model"

File: model/AI/Classes/EnergyConsumptionDescription [hunk 8]
```diff
@@ -0,0 +1,27 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# EnergyConsumptionDescription
+
+## Summary
+
+The Class that helps note down the energy consumption value and the unit in which it is recorded.
```
Comment (bact) [inline model/AI/Classes/EnergyConsumptionDescription:7]: "The Class that" -> "The class that"

**Comment by kestewart** [comment]
@rgopikrishnan91 - can you apply the file extension changes that Gary flagged, and the clarification of  kWh from Art,  and then I'lll apply this.  Merged patch looks good - thanks for doing this.

**Comment by rgopikrishnan91** [comment]
@kestewart @goneall @bact I have made the request changes. Could you guys please review it and merge it?

**Comment by rgopikrishnan91** [comment]
Not sure what the parser is complaining about..I haven't changed anything drastic...

**Comment by goneall** [comment]
Thanks @rgopikrishnan91 

It looks like the validation is failing:

```
0s
Run python /home/runner/work/_temp/spec-parser/main.py -n model
Traceback (most recent call last):
  File "/home/runner/work/_temp/spec-parser/main.py", line [1](https://github.com/spdx/spdx-3-model/actions/runs/8584156125/job/23524144900?pr=697#step:5:1)1, in <module>
    m = Model(cfg.input_dir)
        ^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/_temp/spec-parser/spec_parser/model.py", line 28, in __init__
    self.load(dir)
  File "/home/runner/work/_temp/spec-parser/spec_parser/model.py", line 96, in load
    proptype = self.properties[pname].metadata["Range"]
               ~~~~~~~~~~~~~~~^^^^^^^
KeyError: '/AI/value'
```

I haven't had time to research, but there is probably a reference to a class or property with the name `value` without a corresponding definition.

**Comment by goneall** [comment]
> Yes, @goneall and @rgopikrishnan91 : this PR introduces `EnergyConsumptionDescription` which uses properties `value` and `unit`, but there are no definitions of these properties.

Thanks @zvr - @rgopikrishnan91 - you'll need to add property files for `value` and `unit` to resolve.

**Comment by bact** [comment]
Thank you @rgopikrishnan91 and @zvr.

If so, since:
- the names `value` and `unit` can be very generic even for within the AI domain/namespace; and
- `value` has already been used in the Core for [a property for a generic key-value pair](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Properties/value.md).

could we also rename them to `energyQuantity` and `energyUnit`?

Also shorten `EnergyConsumptionDescriptionUnitType` to `EnergyUnitType` as the type can be used for energy unit of any purpose.

They may all look like this:

----

(1) [MODIFIED] `model/AI/Classes/EnergyConsumptionDescription.md`

```markdown
SPDX-License-Identifier: Community-Spec-1.0

# EnergyConsumptionDescription

## Summary

The class that helps note down the quantity of energy consumption and the unit
used for measurement.

## Description

This class is designed to store energy consumption data, including the quantity
and the unit of measurement.

The energyQuantity property stores the amount of energy consumed,
and the energyUnit property stores the unit used for measurement.

## Metadata

- name: EnergyConsumptionDescription
- Instantiability: Concrete

## Properties

- energyQuantity
  - type: xsd:decimal
  - minCount: 1
  - maxCount: 1
- energyUnit
  - type: EnergyUnitType
  - minCount: 1
  - maxCount: 1
```
----

(2) [NEW] `model/AI/Properties/energyQuantity.md`

```markdown
SPDX-License-Identifier: Community-Spec-1.0

# energyQuantity

## Summary

Represents the energy quantity.

## Description

Provides the quantity information of the energy.

## Metadata

- name: energyQuantity
- Nature: DataProperty
- Range: xsd:decimal
```

----

(3) [NEW] `model/AI/Properties/energyUnit.md`

```markdown
SPDX-License-Identifier: Community-Spec-1.0

# energyUnit

## Summary

Specifies the unit in which energy is measured.

## Description

Provides the unit information of the energy.

## Metadata

- name: energyUnit
- Nature: ObjectProperty
- Range: EnergyUnitType
```

----

(4) [MODIFIED] `model/AI/Vocabularies/EnergyUnitType.md`

(Was: `model/AI/Vocabularies/EnergyConsumptionDescriptionUnitType.md`)

```markdown
SPDX-License-Identifier: Community-Spec-1.0

# EnergyUnitType

## Summary

Specifies the unit of energy consumption.

## Description

List the different acceptable units for measuring energy consumption.

If the unit in which the energy consumption has been recorded
is not listed here, please select "other".

## Metadata

- name: EnergyUnitType

## Entries

- kilowattHour: Kilowatt-hour.
- megajoule: Megajoule.
- other: Any other units of energy measurement. 
```

----

Thanks Gopi again for all the efforts.

**Review by kestewart — APPROVED** [review]
@rgopikrishnan91 - can you apply the fixes that Art, Alexios & Gary suggest?

**Review by kestewart — CHANGES_REQUESTED** [review]
See comments from Art, Alexios & Gary

**Comment by rgopikrishnan91** [comment]
@kestewart @goneall @bact @zvr Hi thanks for all the comments and feedback. I have made the required changes and the parser seems to be happy now. Please let me know if these changes look good.

**Review by kestewart — APPROVED** [review]
Changes look good to me - @bact, @zvr - can you take a look?

**Comment by bact** [comment]
Looks good to me as well. Thank you everyone.

**Review by bact — APPROVED** [review]

**Pull request merged**